pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -199,13 +213,27 @@ class NativeAssetsBuildRunner {
     Uri? resourceIdentifiers,
     required BuildResult buildResult,
   }) async {
+    final loadedUserDefines = await _loadedUserDefines;
+    if (loadedUserDefines?.pubspecErrors.isNotEmpty ?? false) {
+      logger.severe('pubspec.yaml contains errors');
+      for (final error in loadedUserDefines!.pubspecErrors) {
+        logger.severe(error);
+      }
+      return null;
+    }
+    var linkResult = HookResult(
+      dependencies: switch (userDefines?.workspacePubspec) {
+        null => [],
+        final pubspec => [pubspec],
+      },
+    );
+
```

Comment:
Just wondering if its worthwhile to factor this logic out since it seems to be 1:1 repeated from above?